### PR TITLE
Add workflow to update default pack version

### PR DESCRIPTION
**Pull request by importhuman**
This is to create a PR to update the default pack version in github-actions/setup-pack/action.yml when a new pack version is released.
References [update-version-on-release-ga.yml](https://github.com/buildpacks/registry-index/blob/main/.github/workflows/update-version-on-release-ga.yaml). 

Issue: [pack #1244](https://github.com/buildpacks/pack/issues/1244)
Related PR(s): [pack #1263](https://github.com/buildpacks/pack/pull/1263)

### CODE REVIEW

File: .github/workflows/update-version.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+name: update-pack-version
+
+on:
+  repository_dispatch:
+    types:
+      - pack-release
```
Comment (jkutner) [inline .github/workflows/update-version.yml:6]: does this mean the action will be triggered when there's a new release on the pack repo?

Comment (importhuman) [reply .github/workflows/update-version.yml:6]: Yup, wrote a similar workflow for [pack-orb](https://github.com/buildpacks/pack-orb/pull/34)

File: .github/workflows/update-version.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  update-version:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Update pack version of buildpacks/github-actions/setup-pack/action.yml on new pack release
+        uses: actions/checkout@v2
+        run: |
+            NEW_VERSION = $(curl -s -H "Accept: application/vnd.github.v3+json" https://api.github.com/repos/buildpacks/pack/releases/latest | jq .tag_name -r | cut -c 2-)
+            sed -i -z "s/default:     '[0-9]\{1,\}.[0-9]\{1,\}.[0-9]\{1,\}'/default:     '$NEW_VERSION'/2" setup-pack/action.yml
+      - name: Create pull request
+        uses: peter-evans/create-pull-request@v3.10.0
```
Comment (jkutner) [inline .github/workflows/update-version.yml:18]:
```suggestion
        uses: peter-evans/create-pull-request@v3
```

**Review by jkutner — CHANGES_REQUESTED** [review]

File: .github/workflows/update-version.yml (hunk 2 above)
Comment (jkutner) [reply .github/workflows/update-version.yml:18]: matches here https://github.com/buildpacks/github-actions/blob/c9806eb7f23cfe019fdbca8c404d21a520113014/.github/workflows/update-pipeline.yml#L65

**Review by jkutner — APPROVED** [review]

**Comment by importhuman** [comment]
@jkutner the test logs don't seem very informative, what should I do for the failing tests?

**Comment by jkutner** [comment]
@importhuman i think that was just the minimum label enforcer. i've added the labels

**Pull request merged**